lib/circuit_breaker/invalid_transition.rb

```diff
@@ -0,0 +1,15 @@
+class CircuitBreaker::InvalidTransition < RuntimeError
```

Comment:
normally I wouldn't inherit from a runtime error, but it appears that's what aasm does https://github.com/rubyist/aasm/blob/a4bc27cff0d7b7fa5ef61e517555e17b82285254/lib/aasm/errors.rb


```ruby
module AASM
  class InvalidTransition < RuntimeError; end
  class UndefinedState < RuntimeError; end
end

```

Reply:
old code has all the extra methods on it

https://github.com/aasm/aasm/blob/35d6f6843f48760a1e719b507cc832f1a683127f/lib/aasm/errors.rb


```ruby
module AASM

  class UnknownStateMachineError < RuntimeError; end

  class InvalidTransition < RuntimeError
    attr_reader :object, :event_name, :originating_state, :failures

    def initialize(object, event_name, state_machine_name, failures = [])
      @object, @event_name, @originating_state, @failures = object, event_name, object.aasm(state_machine_name).current_state, failures
      super("Event '#{event_name}' cannot transition from '#{originating_state}'.#{reasoning}")
    end

    def reasoning
      " Failed callback(s): #{failures}." unless failures.empty?
    end
  end

  class UndefinedState < RuntimeError; end
  class NoDirectAssignmentError < RuntimeError; end
end

```